### PR TITLE
interface for memberside submitting dev portfolios

### PR DESCRIPTION
**Pull request by airaagrawal1**
### Summary 

This pull request is the first step towards implementing the frontend for Dev Portfolio Form.

- [x] implemented basic interface for member to submit open and reviewed PRs
- [ ] implement ability to add multiple PRs 
- [ ] implement dashboard for members to see the status of dev portfolios submitted in the past
- [ ] connect to backend to test if dropdown for 'select a portfolio assignment' works

Current UI:
Screen Shot 2022-04-16 at 3 34 10 PM

### CODE REVIEW

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Segment, Label, Button, Dropdown } from 'semantic-ui-react';
+import DevPortfolioAPI from '../../../API/DevPortfolioAPI';
+import { Emitters } from '../../../utils';
+import { useSelf } from '../../Common/FirestoreDataProvider';
+// import { DevPortfolio, DevPortfolioSubmission } from '../../../../../common-types/index';
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:6]: remove commented out import

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import styles from './DevPortfolioForm.module.css';
+
+const DevPortfolioForm: React.FC = () => {
+  // When the user is logged in, `useSelf` always return non-null data.
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const userInfo = useSelf()!;
+  console.log(userInfo);
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:13]: remove this console.log

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        member: userInfo,
+        openedPRs: [openPR],
+        reviewedPRs: [reviewedPR],
+        status: 'pending'
+      };
+      requestDevPortfolio(newDevPortfolioSubmission, devPortfolio);
+      Emitters.generalSuccess.emit({
+        headerMsg: 'Dev Portfolio Assignment submitted!',
+        contentMsg: `The leads were notified of your submission and your submission will be graded soon!`
+      });
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:49]: The success message should only be emitted if the request was successfully made

Comment (henry-li-06) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:49]: Also the form should probably be cleared upon a successful submission

**Comment by henry-li-06** [comment]
Will mock data and multiple PR links be included in this PR?

**Comment by airaagrawal1** [comment]
> Will mock data and multiple PR links be included in this PR?

I can do them in a different PR!

**Comment by henry-li-06** [comment]
image
The dev portfolio selector is disappearing for some reason

**Comment by henry-li-06** [comment]
Also mock data should probably be added to this PR as a test plan, otherwise there isn't any way to reasonably test the functionality added in this PR.

**Comment by airaagrawal1** [comment]
> Also mock data should probably be added to this PR as a test plan, otherwise there isn't any way to reasonably test the functionality added in this PR.

The dev portfolio selector should be show up with some mock data now. Because the API isn't set up for accepting responses to the form, clicking the submit button will result in an error for now.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                }}
+              />
+            ) : undefined}
+
+            {devPortfolio ? (
+              <div className={styles.row_direction}>
+                <div>
+                  <Segment>
+                    <h4>{devPortfolio.name}</h4>
+                    <Label>{`deadline: ${devPortfolio.deadline}`}</Label>
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:125]: the deadline will probably need be formatted into a more readable date

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    <div>
+      <Form className={styles.form_style}>
+        <h1>Submit Dev Portfolio Assignment</h1>
+        <p>Submit the recent pull requests you have opened or reviewed here.</p>
+        <div className={styles.inline}>
+          <label className={styles.bold}>
+            Select a Dev Portfolio Assignment <span className={styles.red_color}>*</span>
+          </label>
+          <div className={styles.center_and_flex}>
+            {devPortfolios && !devPortfolio ? (
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:101]: I don't think there needs to be `!devPortfolio`. It's probably better to still display the dropdown even with a dev portfolio selected

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+                <Button
+                  negative
+                  onClick={() => {
+                    setDevPortfolio(undefined);
+                  }}
+                  className={styles.inline}
+                >
+                  Clear
+                </Button>
```
Comment (henry-li-06) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:137]: On the same note, probably don't need this delete button

Comment (airaagrawal1) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx]: Just made the formatting of the dropdown the same as Team Event Credit form. Also changed the type of deadline and earliestValidDate to strings.

**Review by henry-li-06 — APPROVED** [review]
LGTM 🚀

**Pull request merged**